interfaces/apparmor/template.go

```diff
@@ -439,6 +439,11 @@ var templateCommon = `
   # (see 'parallel installs', above)
   /run/snap.@{SNAP_INSTANCE_NAME}/ rw,
   /run/snap.@{SNAP_INSTANCE_NAME}/** mrwklix,
+
+  # Snap-specific lock directory and prerequisite navigation permissions.
+  /run/lock/ r,
+  /run/lock/snap.@{SNAP_INSTANCE_NAME}/ rw,
+  /run/lock/snap.@{SNAP_INSTANCE_NAME}/** mrwklix,
```

Comment:
@zyga, the referenced bug was filed to for microstack's use of /run/lock, but it actually is dealing with /run/lock/LCK.. files, so my comments in https://github.com/snapcore/snapd/pull/8926#discussion_r447981208 apply to this PR. (Ie, I don't think microstack should be using a snap-specific dir for those console lock files).

Looking at other uses for /run/lock, I do see that distro packaging is using the directory for `/run/lock/<application>` which does align with this PR.

In that light, LGTM.